sdk/python/kfp/components/pipeline_channel.py

```diff
@@ -228,6 +228,7 @@ def __init__(
         name: str,
         channel_type: Union[str, Dict],
         task_name: Optional[str],
+        is_artifact_list: bool,
```

Comment:
Pipeline channels must include whether they contain a single artifact or list of artifacts to enable downstream type-checking.